### PR TITLE
Provide clarity on necessity of key property in the data array 

### LINKED ISSUE #434: Lack of documentation around importance of `key` in data array
**Describe the bug**
I have spent about an hour now trying to debug as to why I couldn't get the rows to auto-close when touching other rows. And it finally came down to the fact that my data items did not have a `key` property. They did have a `uuid` property, but that wouldn't map for obvious reasons. So, I finally found the solution after digging through the examples and slowly recreating my own implementation line by line until the bug appeared.

**To Reproduce**
- Open this example: https://snack.expo.io/@jemise111/react-native-swipe-list-view
- Edit this line:
```javascript
    const [listData, setListData] = useState(
        Array(20)
            .fill('')
            .map((_, i) => ({ key: `${i}`, text: `item #${i}` }))
    );
```
to remove the `key` field:
```javascript
    const [listData, setListData] = useState(
        Array(20)
            .fill('')
            .map((_, i) => ({ text: `item #${i}` }))
    );
```
- swipe open multiple rows

Now, the `key` property is obviously necessary for tracking each row. However, this should absolutely be documented somewhere (and if it is, I do apologize but I was unable to find it)

### PR DESCRIPTION
**Pull request by iKrushYou**
Thanks for submitting a pull request!

**Please confirm you have linted your code and fixed any issues:**

* [x] I ran `yarn run fix` on my PR and fixed any formatting issues

**Please provide information on what your PR achieves and any issues that it addresses:**

Fixes #434 
There was previously no documentation surrounding the need for a `key` property in each of the objects in the `data` array. Without this property, the list doesn't function properly: the automatic closing of rows as a result of other interactions breaks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,12 @@ The application under ./SwipeListExample will produce the above example. To run
 ```javascript
 import { SwipeListView } from 'react-native-swipe-list-view';
 
+//... note: your data array objects MUST contain a key property to ensure proper functionality
```
Comment (jemise111) [inline README.md]: Can you amend this to say "MUST contain a key property OR a `keyExtractor` prop..."

File: docs/SwipeListView.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ A List that renders `<SwipeRow />`s
 
 | Prop | Notes | Type | Signature (func) | Default |
 |---|---|---|---|---|
+| `data` | List of objects to be passed into the `renderItem` and `renderHiddenItem` functions. Each item must include a unique `key` property to ensure full functionality. | `array` ||
```
Comment (jemise111) [inline docs/SwipeListView.md]: Same here

File: README.md (hunk 1 above)
Comment (iKrushYou) [reply README.md]: I'm trying to look into the source to see what the usage of `keyExtractor` is

I'm seeing it referenced in `SwipeListView.js` twice as part of the props. However, I don't see it listed in the `propTypes`
I see it being called as `keyExtractor(item, index)` so I can add this to the readme as well as the proptypes if that seems correct

File: docs/SwipeListView.md [hunk 3]
```diff
@@ -6,6 +7,8 @@ A List that renders `<SwipeRow />`s
 
 | Prop | Notes | Type | Signature (func) | Default |
 |---|---|---|---|---|
+| `data` | List of objects to be passed into the `renderItem` and `renderHiddenItem` functions. Each item must include a unique `key` property or `keyExtractor` must be implemented to ensure full functionality. | `array` || 
+| `keyExtractor` | Function to generate `key` value for each row in the list. | `func` |`(item, index) => {}`|
```
Comment (iKrushYou) [inline docs/SwipeListView.md]: I need some advice here on the signature of this function. Wasn't sure how to mimic the format you had for the other ones

**Comment by jemise111** [comment]
Hey @iKrushYou, the `keyExtractor` is a prop of `FlatList` itself, not this library, so I don't think it needs to be added to the docs. If you want you can add a note underneath your comment in the readme directing people to this page: https://reactnative.dev/docs/flatlist#keyextractor

**Comment by iKrushYou** [comment]
@jemise111 ah man.. I did not know that :) 
That makes sense though. I will fix that!

**Comment by iKrushYou** [comment]
Okay, how about now?

**Pull request merged**